File: python-service/rabbitmq.py
import pika
import json
import os

RABBITMQ_URL = os.getenv("RABBITMQ_URL", "amqp://admin:admin123@localhost:5672")

connection = None
channel = None


def connect_rabbitmq():
    global connection, channel

    connection = pika.BlockingConnection(
        pika.URLParameters(RABBITMQ_URL)
    )

    channel = connection.channel()

    channel.queue_declare(queue="prediction_queue", durable=True)
    channel.queue_declare(queue="prediction_result_queue", durable=True)

    print("RabbitMQ Connected")


def get_channel():
    if channel is None:
        raise Exception("RabbitMQ not connected")
    return channel


def close_rabbitmq():
    global connection, channel
    if connection:
        connection.close()

File: Backend/src/rabbitmq-service/config/rabbitmq.js
import amqp from "amqplib";

let connection;
let channel;

export const connectRabbitMQ = async () => {
  connection = await amqp.connect(process.env.RABBITMQ_URL);

  channel = await connection.createChannel();

  await channel.assertQueue("prediction_queue", {
    durable: true,
  });

  await channel.assertQueue("prediction_result_queue", {
    durable: true,
  });

  console.log("RabbitMQ Connected");
};

export const getChannel = () => {
  if (!channel) {
    throw new Error("RabbitMQ belum terkoneksi");
  }

  return channel;
};

export const closeRabbitMQ = async () => {
  await channel?.close();
  await connection?.close();
};
